### PR TITLE
Feature/multiple action modifiers

### LINKED ISSUE #7: Add support for mutiple modifiers & rebuild content
As mentioned in https://github.com/Greenheart/swipeforfuture.com/pull/149#issuecomment-723471244, this repo will need some updates to handle the new data format.
- update the `content-utils` to support generating multiple modifiers.
- rebuild all content and publish a new version.

### PR DESCRIPTION
**Pull request by Greenheart**
Fixes #7 

## What's changed?
- Update `content-utils.ts` to support multiple modifiers
- Update all files in `scripts` folder (including `default` and `donuts` scenarios) to use the new syntax, supporting multiple modifiers.

I've tested everything locally by using method `A)` below

## This PR includes some maintenance updates:
- move `content-utils.ts` to the content scripts since it has been moved. This probably has to be reorganized somehow but for now, everything works as expected.
- Update game submodule to latest version to decrease the module size, thanks to the cleanup PR:s this weekend.
- Also added instructions to README on how to update again in the future, as I imagine this will be a recurring task.

This PR implements the content-side refactoring necessary to support the upcoming game implementation from https://github.com/Greenheart/swipeforfuture.com/pull/149/files

### `A)` Temporary hack to test this PR individually:
1 Checkout the https://github.com/Greenheart/swipeforfuture.com/pull/149/files branch locally and start the game on localhost:3000.
2. Then, you also need to update `ContentTypes.ts` in the game submodule (temporarily) to make sure the content scripts will have support for multiple modifiers. See diff in https://github.com/Greenheart/swipeforfuture.com/pull/149/files for details.
3. Then you can run `npm run dev`, `npm run serve` and test using `localhost:3000?path=http://localhost:5000/default`

### `B)` Or merge game module, update and then test content changes.
1. merge https://github.com/Greenheart/swipeforfuture.com/pull/149/files
2. checkout this PR's branch locally
3. update the game submodule (see instructions included in the README in this PR diff) to get the new updated `ContentTypes.ts`
4. Then you can run `npm run dev` to verify that everything builds, `npm run serve` and finally test using `localhost:3000?path=http://localhost:5000/default`. Remember to update the game to include the latest `master` branch, which should include the merged https://github.com/Greenheart/swipeforfuture.com/pull/149 PR.

Looking forward to hear any feedback!

### CODE REVIEW

File: scripts/content-utils/content-utils.ts [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+export type Modifier = CardActionData["modifiers"][number]
+
+/**
+ * Create a swipeAction by combining a description with one or more modifiers.
+ *
+ * @param description A short text to explain one of the alternatives in a swipe decision.
+ * @param modifiers One or more modifiers that should be applied when the player choose this course of action.
+ */
+export function action(
+    modifiers: Modifier | Modifier[],
```
Comment (Greenheart) [inline scripts/content-utils/content-utils.ts:171]: Perhaps we could even make this optional, to allow `action()` for placeholder actions?

Then we could just set the default value for `modifiers` to `addAction()`

File: scripts/content-utils/content-utils.ts [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+ * @param flags The flags to modify
+ */
+export const replaceModifier = modifier("replace")
+
+/**
+ * Create an modifier factory
+ *
+ * @param type The modifier type to use
+ */
+export function modifier(
```
Comment (Greenheart) [inline scripts/content-utils/content-utils.ts:215]: This new factory adds one modifier at a time, moving the logic to combining multiple modifiers for a single action to the `action()` creator itself.

Really nice to keep the curried functions to get modifier types, while also adding support for multiple modifiers. Took some thought but this felt like a good solution when I used it!

File: scripts/content-utils/content-utils.ts (hunk 1 above)
Comment (Greenheart) [inline scripts/content-utils/content-utils.ts:171]: Also would like to add that allowing single modifiers to be specified without the wrapping array is a big deal for DX. Every character counts for content creation.

It's almost like we want shorthand aliases like `a(aM({}))` which translates to `action(addModifier())`

**Comment by Muthaias** [comment]
Good work. I need some more time to process this one.

Seems to work but I have some input in regards to the order of changes.

I would have preferred us to update `setAction`, `addAction` and `replaceAction` respectively in order to avoid updating content at the same time as util changes. Currently this generates a lot of noise in regards to the diff.

I think it is fine to move forward with this MR once I have had the time to look through it but I need some more time due to all the changes. Moving forward I would suggest trying to take two steps rather than one when possible.

**Comment by Greenheart** [comment]
Thanks! Good point. I actually kept the old `[verb]Action` helpers while I refactored content, but removed it just as I committed.

I'll try to keep it in mind for next time.

File: scripts/content-utils/content-utils.ts (hunk 1 above)
Comment (Muthaias) [reply scripts/content-utils/content-utils.ts:171]: Did you know about a function called addAction? ;)

**Comment by Muthaias** [comment]
Point it to the latest swipeforfuture.com git-version and it's merge time.

**Pull request merged**